packages/massa-web3/src/experimental/errors/base.ts

```diff
@@ -0,0 +1,57 @@
+import { ErrorCodes } from './utils/codes'
+
+export type BaseErrorParameters = {
+  code?: ErrorCodes
+  docsPath?: string
+  metaMessages?: string[]
+} & (
+  | {
+      cause?: never
+      details?: string
+    }
+  | {
+      cause: BaseError | Error
+      details?: never
+    }
+)
+
+export type BaseErrorType = BaseError & { name: 'WalletProviderError' }
+/**
+ * BaseError extends the native Error class to provide additional metadata handling for richer error information.
+ * This class can serve as a base for more specific error types in a wallet or similar system.
+ */
+export class BaseError extends Error {
```

Comment:
Error should be the prefix. this will make the code more readable when listing multiple errors